File: src/test/java/io/mysocialapp/client/java/NotificationTest.java
package io.mysocialapp.client.java;

import io.mysocialapp.client.MySocialApp;
import io.mysocialapp.client.Session;
import io.mysocialapp.client.models.Base;
import io.mysocialapp.client.models.PreviewNotification;
import org.junit.jupiter.api.Test;

/**
 * Created by evoxmusic on 10/05/2018.
 */
class NotificationTest {

    private final static String APP_ID = "u470584465854a728453";
    private final static String EMAIL = "alice.jeith@mysocialapp.io";
    private final static String PASSWORD = "myverysecretpassw0rd";

    private final static Session session = new MySocialApp.Builder()
            .setAppId(APP_ID)
            .build()
            .blockingConnect(EMAIL, PASSWORD);

    @Test
    void listUnreadNotifications() {
        Iterable<PreviewNotification> notifications = session.getNotification().getUnread().blockingList(0, 10);
        notifications.forEach(v -> {
            Base payload = v.getLastNotification().getPayload();
            System.out.println(payload.getDisplayedName());
        });
    }

    @Test
    void listUnreadNotifications_and_consume() {
        Iterable<PreviewNotification> notifications = session.getNotification().getUnread().blockingList(0, 10);
        notifications.forEach(v -> {
            PreviewNotification consumedPreviewNotification = v.blockingConsume();
            Base payload = consumedPreviewNotification.getLastNotification().getPayload();
            System.out.println(payload.getDisplayedName());
        });
    }

    @Test
    void consumeUnreadNotifications() {
        Iterable<PreviewNotification> notifications = session.getNotification().getUnread().blockingListAndConsume();
        notifications.forEach(v -> {
            Base payload = v.getLastNotification().getPayload();
            System.out.println(payload.getDisplayedName());
        });
    }

    @Test
    void listReadNotifications() {
        Iterable<PreviewNotification> notifications = session.getNotification().getRead().blockingList(0, 10);
        notifications.forEach(v -> {
            Base payload = v.getLastNotification().getPayload();
            System.out.println(payload.getDisplayedName());
        });
    }

}

File: src/test/kotlin/io/mysocialapp/client/kotlin/NotificationTest.kt
package io.mysocialapp.client.kotlin

import io.mysocialapp.client.MySocialApp
import io.mysocialapp.client.NotificationCallback
import io.mysocialapp.client.Session
import io.mysocialapp.client.extensions.subscribeAsync
import io.mysocialapp.client.models.*
import org.junit.Test
import java.io.File

/**
 * Created by evoxmusic on 09/05/2018.
 */
class NotificationTest {

    companion object {
        const val APP_ID = "u470584465854a728453"
    }

    private fun getSession(): Session? = MySocialApp.Builder().setAppId(APP_ID).build().blockingConnect("AliceX", "myverysecretpassw0rd")

    @Test
    fun `list unread notifications`() {
        val s = getSession()
        assert(s?.notification?.unread?.blockingStream(100)?.toList() != null)
    }

    @Test
    fun `list read notifications`() {
        val s = getSession()
        assert(s?.notification?.read?.blockingStream(100)?.toList() != null)
    }

    @Test
    fun `consume last unread notification`() {
        val s = getSession()
        val lastNotification = s?.notification?.unread?.blockingStream(1)?.firstOrNull() ?: return
        assert(lastNotification.blockingConsume() != null)
    }

    @Test
    fun `get new event notifications`() {
        val s = getSession()
        val newEventNotifications = s?.notification?.unread?.blockingStream(100)?.toList()
        assert(newEventNotifications != null)
    }

    @Test
    fun `listen for notifications`() {
        val s = getSession()

        s?.notification?.addNotificationListener(object : NotificationCallback {

            override fun onConversationMessage(conversationMessage: ConversationMessage) {
                val message = ConversationMessagePost.Builder().setMessage(conversationMessage.message).build()
                conversationMessage.replyBack(message).subscribeAsync()
            }

            override fun onComment(comment: Comment) {
                val message = CommentPost.Builder().setMessage(comment.message).setImage(File("/tmp/my_profile.png")).build()
                comment.replyBack(message).subscribeAsync()
            }

            override fun onLike(like: Like) {
                val message = CommentPost.Builder().setMessage("Hey [[user:${like.owner?.id}]] thanks for liking this post").build()
                s.newsFeed.blockingGet(like.id!!)?.blockingAddComment(message)
            }

            override fun onNewsFeed(feed: Feed) {
                super.onNewsFeed(feed)
            }

            override fun onMention(feed: Feed) {
                super.onMention(feed)
            }

            override fun onMention(comment: Comment) {
                super.onMention(comment)
            }

            override fun onFriendRequest(user: User) {
                super.onFriendRequest(user)
            }

            override fun onFriend(user: User) {
                super.onFriend(user)
            }

            override fun onEvent(event: Event) {
                super.onEvent(event)
            }
        })


        Thread.sleep(10000000L)
    }

}